datastream-kafka/src/test/java/com/linkedin/datastream/kafka/TestKafkaTransportProvider.java

```diff
@@ -205,6 +205,77 @@ public void testEventWithoutKeyValueAndPartition() throws Exception {
     testEventSend(1, 2, -1, false, false, "test");
   }
 
+  @Test
+  public void testSendMultipleEventsInSingleDatastreamProducerRecord() throws Exception {
+    String metricsPrefix = "test";
+    final int numberOfEvents = 10;
+    String topicName = getUniqueTopicName();
+    final int[] eventsReceived = {0};
+
+    _transportProviderProperties.put(KafkaTransportProviderAdmin.CONFIG_METRICS_NAMES_PREFIX, metricsPrefix);
+    KafkaTransportProviderAdmin provider = new KafkaTransportProviderAdmin("test", _transportProviderProperties);
+
+    String destinationUri = provider.getDestination(null, topicName);
+
+    Datastream ds = DatastreamTestUtils.createDatastream("test", "ds1", "source", destinationUri, 1);
+
+    DatastreamTask task = new DatastreamTaskImpl(Collections.singletonList(ds));
+    TransportProvider transportProvider = provider.assignTransportProvider(task);
+    provider.createTopic(destinationUri, 1, new Properties());
+
+    KafkaTestUtils.waitForTopicCreation(_zkUtils, topicName, _kafkaCluster.getBrokers());
+
+    LOG.info(String.format("Topic %s created with %d partitions and topic properties %s", topicName, 1,
+        new Properties()));
+    DatastreamProducerRecord event =
+        createEventsForSingleDatastreamProducerRecord(topicName, 0, numberOfEvents, true, true);
+    Assert.assertEquals(event.getEvents().size(), numberOfEvents);
+
+    LOG.info(String.format("Trying to send %d events to topic %s", event.getEvents().size(), topicName));
+
+    final Integer[] callbackCalled = {0};
+    transportProvider.send(destinationUri, event, ((metadata, exception) -> callbackCalled[0]++));
```

Comment:
nit: can we also verify the index is as expected?

Reply:
Callbacks may not be returned in the order that the data was sent. What I can do is verify that I receive a callback for every index up to event.getEvents().size(). That should do, right?

Reply:
Sounds good.

Reply:
done